RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -510,18 +500,24 @@ class PumpOpsSynchronous {
                 timestamp.timeZone = pump.timeZone
 
                 if let date = timestamp.date?.addingTimeInterval(timeAdjustmentInterval) {
-                    if date.timeIntervalSince(startDate) < -eventTimestampDeltaAllowance {
-                        NSLog("Found event at (%@) to be more than %@s before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
-                        return (events: events, hasMoreEvents: false)
-                    } else if date.timeIntervalSince(timeCursor) > eventTimestampDeltaAllowance {
-                        NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
-                        return (events: events, hasMoreEvents: false)
-                    } else {
-                        if (date.compare(startDate) != .orderedAscending) {
-                            timeCursor = date
-                        }
-                        events.insert(TimestampedHistoryEvent(pumpEvent: event, date: date), at: 0)
+                    
+                    let possibleCancelBecauseOfDate = !event.isDelayedAppend(withPumpModel: pumpModel)
```

Comment:
We're not 'cancelling' the operation because of date.  When we reach a nonDelayedAppend event that is older than start date, we've finished successfully.  The only time we cancel is when we hit a pump time loss.